### PR TITLE
Add comments for error cases in status functions

### PR DESCRIPTION
**Pull request by hqhq**
Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

### CODE REVIEW

File: libcontainer/container_linux.go [hunk 1]
```diff
@@ -1043,9 +1043,6 @@ func (c *linuxContainer) isRunning() (bool, error) {
 	}
 	// return Running if the init process is alive
 	if err := syscall.Kill(c.initProcess.pid(), 0); err != nil {
-		if err == syscall.ESRCH {
-			return false, nil
-		}
```
Comment (dqminh) [inline libcontainer/container_linux.go]: this is a known error from kill, means that the process does not exist anymore (in waiting state) for example, so i think the original code makes sense.

File: libcontainer/container_linux.go [hunk 2]
```diff
@@ -1054,9 +1051,6 @@ func (c *linuxContainer) isRunning() (bool, error) {
 func (c *linuxContainer) isPaused() (bool, error) {
 	data, err := ioutil.ReadFile(filepath.Join(c.cgroupManager.GetPaths()["freezer"], "freezer.state"))
 	if err != nil {
-		if os.IsNotExist(err) {
-			return false, nil
-		}
```
Comment (dqminh) [inline libcontainer/container_linux.go]: iirc i think this is where we check if the host has freezer cgroup, if not the application obviously cant be frozen, hence we can resolved the pause state here.

**Comment by cyphar** [comment]
I agree with @dqminh, it looks like they are there because "errors" from the OS are really just "this already happened" states. We could add comments though.

**Comment by hqhq** [comment]
@dqminh Thanks, I think you are right, I've updated the PR to add comments instead.

**Comment by crosbymichael** [comment]
LGTM

**Comment by cyphar** [comment]
LGTM

**Pull request merged**